src/main/scala/ai/starlake/job/sink/pubsub/PubSubSource.scala

```diff
@@ -0,0 +1,235 @@
+package ai.starlake.job.sink.pubsub
+
+import ai.starlake.job.sink.DataFrameTransform
+import ai.starlake.utils.Utils
+import com.google.cloud.pubsub.v1.stub.{GrpcSubscriberStub, SubscriberStubSettings}
+import com.sun.net.httpserver.HttpExchange
+import com.google.cloud.pubsub.v1.{AckReplyConsumer, MessageReceiver, Subscriber}
+import com.google.pubsub.v1.{
+  AcknowledgeRequest,
+  ProjectSubscriptionName,
+  PubsubMessage,
+  PullRequest
+}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.ai.starlake.http.HttpSourceProxy
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.streaming.{LongOffset, Offset, SerializedOffset, Source}
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.apache.spark.unsafe.types.UTF8String
+import com.typesafe.scalalogging.StrictLogging
+
+import java.util.concurrent.TimeoutException
+import scala.jdk.CollectionConverters.{IterableHasAsJava, ListHasAsScala}
+import java.util.concurrent.atomic.AtomicBoolean
+import scala.collection.mutable.ListBuffer
+
+case class PubSubPayload(data: String)
+class PubSubSource(sqlContext: SQLContext, parameters: Map[String, String])
+    extends HttpSourceProxy
+    with Source
+    with StrictLogging {
+
+  private val projectId = parameters.getOrElse("gcpProjectId", "not_found_in_parameters")
+  private val subscriptionId = parameters.getOrElse("gcpSubscriptionId", "not_found_in_parameters")
+  assert(
+    projectId != "not_found_in_parameters",
+    s"Parameter gcpProjectId is mandatory : ${parameters}"
+  )
+  assert(
+    subscriptionId != "not_found_in_parameters",
+    "Parameter gcpSubscriptionId is mandatory"
+  )
+  private val awaitTimeout = sys.env.get("AWAIT_TIMEOUT").map(_.toInt).getOrElse(600)
```

Comment:
env vars should start with SL_ and be defined in the reference.conf file under the section 
```
pubsub {
  awaitTimeoutSeconds = 600
  awaitTimeoutSeconds = ${?SL_PUBSUB_AWAIT_TIMEOUT
  ...
}
```